libraries/standard/http/utest/CMakeLists.txt

```diff
@@ -9,11 +9,11 @@ set(project_name "http")
 
 # list the files to mock here
 list(APPEND mock_list
-            ""
+            ${MODULES_DIR}/standard/http/third_party/http_parser/http_parser.h
```

Comment:
Why are parser files not part of `HTTP_SOURCES` in `filepaths.cmake`? Do we need a separate variable `HTTP_PARSER_SOURCES` and same for include paths in `filepaths.cmake`?

Reply:
I'm not so sure about adding `HTTP_PARSER_SOURCES` to `filepaths.cmake`. I was told by @leegeth to only include sources that are in both AFR and C SDK but share the same directory. AFR already has http_parser under `libraries/3rdparty`, so the directory would be different.

Reply:
For SSoT project, the `C-SDK` will be the SSoT for `HTTP`, thus, I don't think we need to be constrained by the directory structure of `HTTP` on the current version of AFR (as HTTP will be removed from it in the future, and only sourced from C-SDK)

Reply:
Source files and include directories for libraries(in this case http lib) will be used by build systems in repositories in which we use those libraries for SSoT. I have a question, how are we planning to build http_parser? Is it going to be added as a library and then a dependency to http lib or we include those as part of http library source itself?
If the option is the first one, for consistency, we can use filePaths.cmake for http_parser and use it for building http parser.

Reply:
Ideally, just like `cmock` and `unity`, `http-parser` should have its own library target defined in its own directory

Reply:
Creation of a separate library for `http-parser` can be done in a separate PR. This PR was primarily meant for fixing issues in the build system blocking development